src/telliot_core/contract/contract.py

```diff
@@ -156,17 +189,33 @@ async def write(
     async def write_with_retry(
```

Comment:
The `write_with_retry` function is pretty confusing to start, and now even more so, and doesn't have any documentation.  It makes me nervous using it.  At the very least, we should document the intent/goal, what situations it handles, when it raises gas prices, etc.  It can have a significant impact on profit.